debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorTask.java

```diff
@@ -79,6 +79,7 @@ public synchronized void start(Map<String, String> props) {
 
                 // First check if db history is available
                 if (!taskContext.historyExists()) {
+                    taskContext.initializeHistoryStorage();
```

Comment:
Tiny change, but perhaps move this after the isBinlogAvailable() check below, since that is the definite point at we would be starting the snapshot.

Reply:
I'd disagree here - this is more philosophical question - but I prefer the initializiton of storage (topic) immdiately after it is deteced it is not present

Reply:
The thing is that we may end up having created a topic which isn't needed, as an exception is thrown in the next step.

Reply:
@gunnarmorling ok, Moved later to create the topic only when needed